src/api/app/components/build_result_for_architecture_component.rb

```diff
@@ -0,0 +1,67 @@
+class BuildResultForArchitectureComponent < ApplicationComponent
+  attr_accessor :result, :project, :package
+
+  def initialize(result, project, package)
+    super
+
+    @result = result
+    @project = project
+    @package = package
+  end
+
+  private
+
+  def result_code
+    result.code
+  end
+
+  def help
+    help = "<p><strong>Package build ( #{build_status_icon} #{result_code} ):</strong> #{build_status_help}</p>"
+    help += "<p><u>Details</u>: #{build_status_details}</p>" if build_status_details.present?
+    help += "<p><strong>Repository status ( #{repository_status_icon} #{result.state} ): </strong>#{repository_status_help}</p>"
+    help
+  end
+
+  def build_status_help
+    Buildresult.status_description(result_code)
+  end
+
+  def build_status_details
+    result.details
+  end
+
+  def repository_status_help
+    result.is_repository_in_db ? helpers.repository_info(result.state) : 'This result is outdated'
+  end
+
+  def build_status_icon_with_link
+    return build_status_icon if result_code.in?(['-', 'unresolvable', 'blocked', 'excluded', 'scheduled'])
+
+    link_to(package_live_build_log_path(project: project.to_s, package: package,
+                                        repository: result.repository, arch: result.architecture), rel: 'nofollow') do
+      capture { build_status_icon }
+    end
+  end
+
+  def build_status_icon
+    tag.i(class: "fa #{helpers.build_status_icon(result_code)} text-gray-500")
+  end
+
+  def repository_status_icon
+    helpers.repository_status_icon(status: result.state)
+  end
+
+  def status_border_color(status)
+    return 'border-warning' if Buildresult.in_progress_status?(status)
```

Comment:
I believe you can initialize BuildResult with the status and then query that buildresult about it's several states, instead of feeding the status every time. You know:
```
buildresult = BuildResult.new(status)
buildresult.in_progress?
buildresult.successful?
etc
```
Otherwise, there is no reason to use a class, just a `BuildResult` module with methods inside.

Reply:
I've just pushed the changes

Reply:
I added a card for that: https://trello.com/c/6cqcj2DI/54-replace-class-methods-by-instance-methods-in-model-buildresult